### PR TITLE
Fix issue with chip-repl yamltest running TestClusterComplexTypes.yaml

### PR DESCRIPTION
**Pull request by tehampson**
Problem: chip-repl yamltest would fail TestClusterComplexTypes.yaml for the following two reasons:
* We were not providing the `timedRequestTimeoutMs`
* We had no way to force a busy wait loop in the eventloop after the SendCommand/WriteAttribute

Solution:
* Provide `timedRequestTimeoutMs`
* Pass along busy wait time in ms to occur after sending out the command so that the client event loop is suspended

### CODE REVIEW

File: src/controller/python/chip/clusters/attribute.cpp [hunk 1]
```diff
@@ -370,6 +370,16 @@ PyChipError pychip_WriteClient_WriteAttributes(void * appContext, DeviceProxy *
     client.release();
     callback.release();
 
+    if (busyWaitMs) {
+        auto durationInMs = chip::System::Clock::Milliseconds32(busyWaitMs);
+        auto & clock = chip::System::SystemClock();
+        auto start   = clock.GetMonotonicTimestamp();
+        while (clock.GetMonotonicTimestamp() - start < durationInMs)
+        {
+            // nothing to do.
```
Comment (bzbarsky-apple) [inline src/controller/python/chip/clusters/attribute.cpp:379]: Do you need to busy-wait, or can you just `usleep()`?

File: src/controller/python/chip/clusters/command.cpp [hunk 2]
```diff
@@ -161,6 +163,16 @@ PyChipError pychip_CommandSender_SendCommand(void * appContext, DeviceProxy * de
     sender.release();
     callback.release();
 
+    if (busyWaitMs) {
+        auto durationInMs = chip::System::Clock::Milliseconds32(busyWaitMs);
+        auto & clock = chip::System::SystemClock();
+        auto start   = clock.GetMonotonicTimestamp();
+        while (clock.GetMonotonicTimestamp() - start < durationInMs)
```
Comment (bzbarsky-apple) [inline src/controller/python/chip/clusters/command.cpp:170]: Again, could this just sleep?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/controller/python/chip/clusters/attribute.cpp (hunk 1 above)
Comment (tehampson) [reply src/controller/python/chip/clusters/attribute.cpp]: That works! Thanks

File: src/controller/python/chip/ChipDeviceCtrl.py [hunk 3]
```diff
@@ -867,10 +867,10 @@ async def SendCommand(self, nodeid: int, endpoint: int, payload: ClusterObjects.
                 EndpointId=endpoint,
                 ClusterId=payload.cluster_id,
                 CommandId=payload.command_id,
-            ), payload, timedRequestTimeoutMs=timedRequestTimeoutMs, interactionTimeoutMs=interactionTimeoutMs).raise_on_error()
+            ), payload, timedRequestTimeoutMs=timedRequestTimeoutMs, interactionTimeoutMs=interactionTimeoutMs, busyWaitMs=busyWaitMs).raise_on_error()
         return await future
 
-    async def WriteAttribute(self, nodeid: int, attributes: typing.List[typing.Tuple[int, ClusterObjects.ClusterAttributeDescriptor, int]], timedRequestTimeoutMs: int = None, interactionTimeoutMs: int = None):
+    async def WriteAttribute(self, nodeid: int, attributes: typing.List[typing.Tuple[int, ClusterObjects.ClusterAttributeDescriptor, int]], timedRequestTimeoutMs: int = None, interactionTimeoutMs: int = None, busyWaitMs: int = None):
```
Comment (andy31415) [inline src/controller/python/chip/ChipDeviceCtrl.py:873]: type should be `Optional[int]` or `int | None` or `Union[int, None]` if the default is None. It seems other arguments fall under that as well.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**